src/docs/arch-block.dox

```diff
@@ -1,9 +1,132 @@
 /*! @arch_page arch-block Block Manager
 
-The Block Manager manages the reading and writing of disk blocks
-in WiredTiger.  It does compression and encryption when these
-are configured.
+The WiredTiger block manager subsystem manages the reading and writing of data
+from the disk. It is designed to facilitate high performance, economic use of
+disk space and customisability.
+
+@section block What is a block?
+
+The block manager API introduces the concept of a "block" which represents a
+chunk of data that is stored on the disk and operated on as a single atomic
+unit. Each WiredTiger data file (any file in the home directory with the \c .wt
+suffix) is made up of these blocks.
+
+Since a data file can contain page images from multiple reconciliations of the
+same page, each block is associated with a single checkpoint. The API is
+designed in a way that allows users to load a checkpoint into the block manager
+and then read the blocks from that checkpoint as if they were a contiguous
+stream of chunks in the file (in reality, blocks from different checkpoints can
+be interleaved).
+
+@section block_usage How WiredTiger uses the block manager
+
+Whenever a page is read into the cache, WiredTiger will read its corresponding
+block from the data file. This block contains what's referred to as the
+"page image" in other subsystems and has the newest versions of data for each
+key.
+
+Conversely, when a page is reconciled whether that is due to a page eviction or
+by a checkpoint, a new page image is constructed and it is written as a new
+block in the data file. In the case of checkpoint, the newly written block is
+bound to the current checkpoint and in the case of eviction, it is associated
+with the next checkpoint.
+
+@section block_implementation How the block manager is implemented
+
+We've already discussed the concept of a block and how the data file's on-disk
+representation is divided into smaller blocks that are each associated with a
+checkpoint.
+
+In this section, we'll dive into the file layout in more detail and talk about
+some of the core data structures used as part of block allocation.
+
+@subsection block_layout Block layout
+
+Each block consists of a page header, a block manager header followed by the
+actual page image.
```

Comment:
I'd say "a page header, followed by a block manager header, followed by ...", so it's clear there are 3 distinct pieces.

Reply:
3536114